### ISSUE TITLE
Future of sessions

### ISSUE THREAD

**Issue by hongaar** [vorpaljs-reforged/vorpal#29]
There's a lot of code in this project dealing with sessions, sending events up/downstream, server/client logic, maintaining queues, etc. There's also some direct references to `vantage`.

I'm under the impression the session related code is primarily to support vantage which is built on top of vorpal. No article on the wiki nor the website mention any of these APIs.

For me, this secondary use-case complicates working with the code and distracts from the primary use-case (local cli framework).

Is anybody aware of any use-cases for the client/server/session code, or actively using it in a project?
Vantage itself is still used but the downloads on npm is ~150 / month vs ~55.000 / month for vorpal. 

To simplify the codebase, and allow for more straight-forward refactoring and testing, I propose to drop all session related code (but not before we established a fully typed and tested codebase).

Maybe we release v2 to establish a ts port with feature parity, and a v3 soon thereafter where we drop all deprecated features?  

Curious to hear how others think about this.

**Comment by AdrieanKhisbe** [comment]
I totaly agree. Sessions via vantage does not seems to bé used in pratice, and the code overhead is huge.

We might even maybe drop it right now :thinking: 
mainly to simplify API/types finalisation.
(Both for v2 And for v3, And both for us And our user)

Also it would be still be possible to reimplement it if ever thé need raises.

**Comment by hongaar** [comment]
I think for now it would be easier to type everything first (even though it would take slightly longer), and then remove it so we can be more confident when refactoring.

**Comment by Ore4444** [comment]
I agree with @hongaar regarding order of operations. We could remove it before v1.
Might be useful in the future as vorpal-reforged/vorpal-session
